platform/darwin/src/MGLLoggingConfiguration_Private.h

```diff
@@ -11,6 +15,20 @@
         [[MGLLoggingConfiguration sharedConfiguration] logCallingFunction:function functionLine:line messageType:type format:(message), ##__VA_ARGS__]; \
     } \
 }
+#define MGLAssert(expression, message, ...) \
+    __extension__({ \
+        if (__builtin_expect(!(expression), 0)) { \
+            MGLLogFault(message, ##__VA_ARGS__); \
+        } \
+        NSAssert(expression, message, ##__VA_ARGS__); \
+    })
+#define MGLCAssert(expression, message, ...) \
+    __extension__({ \
+        if (__builtin_expect(!(expression), 0)) { \
+            MGLLogFault(message, ##__VA_ARGS__); \
+        } \
+        NSCAssert(expression, message, ##__VA_ARGS__); \
+    })
```

Comment:
We should modify this and `MGLAssert` to avoid a potential re-evaluation of `expression`, so something like:
```
#define MGLCAssert(expression, message, ...) \
    __extension__({ \
        __typeof__(expression) exp = expression; \
        if (__builtin_expect(!(exp), 0)) { \
            MGLLogFault(message, ##__VA_ARGS__); \
        } \
        NSCAssert(exp, message, ##__VA_ARGS__); \
    })
```